src/sql/SqlService.ts

```diff
@@ -311,12 +318,32 @@ export class SqlServiceImpl implements SqlService {
                 this.connectionManager.getClientUuid()
             );
 
-            return this.invocationService.invokeOnConnection(connection, requestMessage).then(clientMessage => {
-                SqlServiceImpl.handleExecuteResponse(clientMessage, res);
-                return res;
+            return this.invocationService.invokeOnConnection(
+                connection, requestMessage, SqlExecuteCodec.decodeResponse
+            ).then(response => {
+                SqlServiceImpl.handleExecuteResponse(response, result);
+                // Try to deserialize the first row for compact schema fetching if lazy deserialization won't be used.
```

Comment:
What about putting this schema fetching logic in the `SqlResult.next()`? While we are not returning `SqlRow` objects to the user, eager deserialization will happen and we will be able to fetch the schemas. If you don't like this idea as well, we should describe how to fix the error to the user.

```
message = 'You tried to deserialize an SQL row which includes a compact serializable object, however '
                        + 'the schema for that object is not known by the client. The client won\'t fetch the schema of '
                        + 'the field because of lazy deserialization support. In order not to get this error, you can change '
                        + 'your mapping to use the fields of the compact object. SQL\'s lazy deserialization support may '
                        + 'be removed in the future, after that you will no longer get this error.'
```

Using `map.get()` to fetch the schame is not reliable as well if the map includes different compact objects under different keys.

Reply:
I didn't get `Using map.get() to fetch the schame is not reliable as well if the map includes different compact objects under different keys.` part

Reply:
I meant we can tell the user that they can fetch the schema via APIs like `map.get()`. However, if different keys include different compact objects it is impossible to fetch all schemas one by one. Maybe via `getAll` etc.